pipelines/release/release.yaml

```diff
@@ -444,6 +456,61 @@ jobs:
             EOF
             echo "merging with cluster values..."
             spruce merge ./platform/charts/gsp-cluster/values.yaml ./overrides.yaml | tee -a cluster-values/values.yaml
+  - task: pick-platform-version
+    image: concourse-task-toolbox
+    config:
+      platform: linux
+      inputs:
+      - name: full-gsp
+      - name: platform
+      - name: service-operator-source
+      - name: concourse-task-toolbox-source
+      - name: concourse-operator-source
+      - name: concourse-github-resource-source
+      - name: concourse-harbor-resource-source
+      outputs:
+      - name: platform-version
+      run:
+        path: /usr/bin/python3
+        args:
+        - -c
+        - |
+          import os
+          import subprocess
+          import sys
+
+          os.mkdirs("platform-version", exist_ok=True)
+
+          print("Picking platform version...")
+          partial_repos = [
+              "platform",
+              "service-operator-source",
+              "concourse-task-toolbox-source",
+              "concourse-operator-source",
+              "concourse-github-resource-source",
+              "concourse-harbor-resource-source"
+          ]
+          hashes = set(map(lambda r: open(f"{r}/.git/ref").read(), partial_repos))
+
+          proc = subprocess.Popen(
+              ['git', 'log', '--format=%H'],
```

Comment:
```suggestion
              ['git', 'log', '--format=%H', '-n', '300'],
```

Reply:
I'm not keen on the idea of putting an arbitrary limit in.